src/Jackalope/Transport/DoctrineDBAL/Client.php

```diff
@@ -1539,8 +1539,13 @@ private function getSystemIdForNode($identifier, $workspaceName = null)
         if (UUIDHelper::isUUID($identifier)) {
             $query = 'SELECT id FROM phpcr_nodes WHERE identifier = ? AND workspace_name = ?';
         } else {
-            if ($this->getConnection()->getDriver() instanceof \Doctrine\DBAL\Driver\PDOMySql\Driver) {
-                $query = 'SELECT id FROM phpcr_nodes WHERE path COLLATE utf8_bin = ? AND workspace_name = ?';
+            $params = $this->conn->getParams();
+            $charset = isset($params['charset']) ? $params['charset'] : 'utf8';
```

Comment:
yes, i would say so. have a setter, and a get method that does the code you added above and below, so that this code can just say if mysql and not binary, do collate with $this->getCaseSensitiveEncoding()